src/infra/core/tests/benches/parallel_simple_transfer_protocol.rs

```diff
@@ -0,0 +1,205 @@
+    SyncServiceImpl,
+};
+use opendatafabric::*;
+use url::Url;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+async fn setup_dataset(
+    tmp_workspace_dir: &Path,
+    dataset_alias: Arc<DatasetAlias>,
+    ipfs: Option<(IpfsGateway, IpfsClient)>,
+) -> (Arc<dyn SyncService>, Arc<DatasetRepositoryLocalFs>) {
+    let (ipfs_gateway, ipfs_client) = ipfs.unwrap_or_default();
+
+    let catalog = dill::CatalogBuilder::new()
+        .add::<EventBus>()
+        .add::<DependencyGraphServiceInMemory>()
+        .add_value(ipfs_gateway)
+        .add_value(ipfs_client)
+        .add_value(CurrentAccountSubject::new_test())
+        .add_builder(
+            DatasetRepositoryLocalFs::builder()
+                .with_root(tmp_workspace_dir.join("datasets"))
+                .with_multi_tenant(false),
+        )
+        .bind::<dyn DatasetRepository, DatasetRepositoryLocalFs>()
+        .add_builder(
+            RemoteRepositoryRegistryImpl::builder().with_repos_dir(tmp_workspace_dir.join("repos")),
+        )
+        .bind::<dyn RemoteRepositoryRegistry, RemoteRepositoryRegistryImpl>()
+        .add::<auth::DummyOdfServerAccessTokenResolver>()
+        .add::<DatasetFactoryImpl>()
+        .add::<SyncServiceImpl>()
+        .add::<DummySmartTransferProtocolClient>()
+        .add::<auth::AlwaysHappyDatasetActionAuthorizer>()
+        .build();
+
+    let sync_svc = catalog.get_one::<dyn SyncService>().unwrap();
+    let dataset_repo = catalog.get_one::<DatasetRepositoryLocalFs>().unwrap();
+
+    // Add dataset
+    let snapshot = MetadataFactory::dataset_snapshot()
+        .name(&dataset_alias.dataset_name)
+        .kind(DatasetKind::Root)
+        .push_event(MetadataFactory::set_data_schema().build())
+        .build();
+
+    let _ = dataset_repo
+        .create_dataset_from_snapshot(None, snapshot)
+        .await
+        .unwrap()
+        .head;
+
+    // Generate initial 70 blocks in dataset
+    for _ in 1..70 {
+        let _ = DatasetTestHelper::append_random_data(dataset_repo.as_ref(), &*dataset_alias, true)
+            .await;
+    }
+
+    (sync_svc, dataset_repo)
+}
+
+async fn do_test_sync(
+    sync_svc: Arc<dyn SyncService>,
+    dataset_alias: Arc<DatasetAlias>,
+    pull_repo_url: &DatasetRefRemote,
+    push_repo_url: &DatasetRefRemote,
+    dataset_repo: Arc<DatasetRepositoryLocalFs>,
+) {
+    let _push_res = sync_svc
+        .sync(
+            &dataset_alias.as_any_ref(),
+            &push_repo_url.as_any_ref(),
+            SyncOptions::default(),
+            None,
+        )
+        .await;
+
+    let _pull_res = sync_svc
+        .sync(
+            &pull_repo_url.as_any_ref(),
+            &dataset_alias.as_any_ref(),
+            SyncOptions::default(),
+            None,
+        )
+        .await;
+
+    // Generate additional 70 blocks in dataset to make sure next iteration will be
+    // the same as previos
+    for _ in 1..70 {
```

Comment:
Number 70 needs to be an algorithm argument. 
It's randomly written over different test routines, and if we change the value in once place, it's easy to forget to change it in another.